tests/UI/pages/BO/advancedParameters/email/index.js

```diff
@@ -208,6 +224,92 @@ class Email extends BOBasePage {
   async isLogEmailsTableVisible(page) {
     return this.elementVisible(page, this.emailGridPanel, 1000);
   }
+
+  /**
+   * Get pagination label
+   * @param page
+   * @return {Promise<string>}
+   */
+  getPaginationLabel(page) {
+    return this.getTextContent(page, this.paginationLabel);
+  }
+
+  /**
+   * Select pagination limit
+   * @param page
+   * @param number
+   * @returns {Promise<string>}
+   */
+  async selectPaginationLimit(page, number) {
+    await Promise.all([
+      this.selectByVisibleText(page, this.paginationLimitSelect, number),
+      page.waitForNavigation({waitUntil: 'networkidle'}),
+    ]);
+
+    return this.getPaginationLabel(page);
+  }
+
+  /**
+   * Click on next
+   * @param page
+   * @returns {Promise<string>}
+   */
+  async paginationNext(page) {
+    await this.clickAndWaitForNavigation(page, this.paginationNextLink);
+
+    return this.getPaginationLabel(page);
+  }
+
+  /**
+   * Click on previous
+   * @param page
+   * @returns {Promise<string>}
+   */
+  async paginationPrevious(page) {
+    await this.clickAndWaitForNavigation(page, this.paginationPreviousLink);
+
+    return this.getPaginationLabel(page);
+  }
```

Comment:
There are many methods duplicated for migrated and not migrated pages, and these methods use different selectors. Also, `BOBasePage` contains only basic methods.
@boubkerbribri I think we have already chosen to not put these methods in BOBasePage, WDYT ?

Reply:
This needs more thinking, It's not just this method but almost all methods in listing pages.

What I'm sure about  : the BoBasePage is not the right place to do it because BoBasePage only contains functions that can be used everywhere in BO. Adding this functions there, will break this rule.

The solution (maybe) : another level of inheritance could be added for listing pages (2 base pages : 1 for migrated pages, and the other for legacy ones). But it will need a lot of time.

Reply:
Anyway, I think this is not the scope here, and we will discuss solutions before going further.